java/opendcs/src/main/java/decodes/cwms/resevapcalc/EvapUtilities.java

```diff
@@ -0,0 +1,716 @@
+    {            
+        final double[] rs = {0.135,0.16,1.00,3.00,10.0,30.0,100.0,300.0,1000.0};
+        final double[] at = {0.177,1.376,1.376,1.026,1.625,4.661,34.904,1667.19,5.88E5};
+        final double[] bt = {0.0,0.929,0.929,-0.599,-1.018,-1.475,-2.067,-2.907,-3.935};
+        final double[] aq = {0.292,1.808,1.808,1.393,1.956,4.994,30.709,1448.68,2.98E5};
+        final double[] bq = {0.0,0.826,0.826,-0.528,-0.870,-1.297,-1.845,-2.682,-3.616};      
+        
+        double gu = gnu/ustar;  // Viscous length scale
+        
+        int idx = 8;
+        for ( int i=0; i<9; i++ )
+        {
+            if ( rstar < rs[i] )
+            {
+                idx = i;
+                break;
+            }
+        }
+        
+        //  Compute ZT and ZQ in RSTAR range identified by i
+        zt.d = gu*at[idx]*(  Math.pow( rstar, bt[idx]) );
+        zq.d = gu*aq[idx]*(  Math.pow( rstar, bq[idx]) );
+    }
+    
+    /**
+     * This subroutine computes the momentum roughness over snow (Z0) based
+     * on observations on Ice Station Weddell.  It recognizes three regimes:
+     * An aerodynamically smooth regime, where Z0 goes as GNU/USTAR; a
+     * drifting-snow regime, where Z) goes as (USTAR**2)/G; and a
+     * statistically determined intermediate regime that allows increased
+     * roughness when the flow is transvers to the sastrugi.
+     * 
+     * @param ustar
+     * @param gnu
+     * @return 
+     */
+    public static double drift_snow( double ustar, double gnu )
+    {            
+        double z0_smooth = smooth(ustar, gnu);     // Aerodynamically smooth part
+        double z0_drift = (0.065/Constants.CONST_G)*(ustar*ustar);     //Drifting snow part
+        double utmp = ((ustar-0.25)/0.052);
+        double z0_stat = 1.13E-3* Math.exp( -(utmp*utmp));   //Statistical intermediate partt
+        
+        // Overall roughness length
+        double z0 = z0_smooth + z0_stat + z0_drift;
+        return z0;
+    }
+    
+    /**
+     * Computes the scalar roughnesses for temperature
+     * (ZT) and humidity (ZQ) over now using the Andreas (1987)
+     * theoretical model.
+     * 
+     * All units are mks.
+     * The predictive equation is
+     * 	  ln(ZS/Z0)  =  B0  +  B1*ln(RSTAR)  +  B2*(ln(RSTAR)**2)
+     * 
+     * @param ustar  - the friction velocity
+     * @param z0     - the momentum roughness length.
+     * @param rstar  - is the roughness Reynolds number.
+     * @param zt     - (Output) scalar roughnesses for temperature
+     * @param zq     - (Output) scalar roughnesses for humidity
+     */
+    public static void andreas_87( double ustar, double z0, double rstar,
+            DoubleContainer zt, DoubleContainer zq )
+    {            
+        final double[] rs = {0.135,2.5,1000.};       
+                
+        final double[][] bt = 
+        { {1.250,0.0,0.0},
+          {0.149,-0.550,0.0}, 
+          {0.317,-0.565,-0.183 } };           
```

Comment:
I think these might read better with the first and last curly each on their own line.